### ISSUE TITLE
Very long response times

### ISSUE THREAD

**Issue by jcjdammers** [glk1001/plugin.audio.spotify#8]
Hi,

I am experiencing very long response times navigating through the menus. See below. I am using v1.3.12 of the addon.

Startup the addon: 10s
Selecting "My music": 25s
Selecting "Playlists": 25s
Selecting one of the playlists from the list: 45s
Selecting "Albums":  40s
Selection one of the albums from the list: 50s
Selecting "Artists":  40s
Selecting one of the artists from the list: 70s

These response times are very annoying and I really hope you can do something to improve them.
The playing of songs/playlist themselves works fine.

I am running LibreELEC 12.0.1 with Kodi (Omega) v21.1 (for Generic PC).

Thanks,
Jeroen.

**Comment by glk1001** [comment]
Yes, these times seem too long. There is playlist caching so loading a
second time should be a lot quicker. Can you give me an idea of how many
tracks are in the 45s playlist?

**Comment by jcjdammers** [comment]
That playlist contains 169 tracks.

https://open.spotify.com/playlist/4OHlLEj27Q23cAK92IpYVZ

**Comment by jcjdammers** [comment]
@glk1001 ,
Any chance of improving the response times?
Would be much appreciated.

**Comment by glk1001** [comment]
Hi, Sorry for the delay in getting back (Christmas). I've added a new version '1.3.13-test' at https://glk1001.github.io/ that logs the number of items in cached and uncached retrievals. If you install that and check the Kodi log we might be able to figure out what's going on

**Comment by jcjdammers** [comment]
Hi,
I have installed the new version, but it is not showing any playlist or album anymore.
I see **"TypeError: 'NoneType' object is not subscriptable"** errors in the Kodi log-file:
https://paste.kodi.tv/kucizaviqo

**Comment by glk1001** [comment]
Sorry, I've updated version '1.3.13-test' with logging fixes. Try again.

**Comment by jcjdammers** [comment]
This one is working :-)
Besides the logging, did you make other changes as well? 
I tried a couple of playlists and albums but the response times are good now. Much better then before.
See the log:
https://paste.kodi.tv/guwewudara

**Comment by glk1001** [comment]
Ah, yes. While adding the logging I saw something that also looked odd. The
default retrieval of albums for artists was to get every possible track for
the artist: album tracks, singles, and any albums that the artist appeared
on. I think the 'appeared on' would add a large amount of tracks for some
artists. So as well as the logging change, I changed the default back to
singles and albums retrieval. I think this would explain your problem and
why it's OK now. This bug probably goes back to plugin version 1.3.11,
where I added code to handle a change that Spotify made in their API
affecting default track retrieval.

Sorry I took so long to sort it out.

**Comment by jcjdammers** [comment]
Aha that would probably explain it. Thanks for looking into it.
Now that it is OK, maybe you can create a clean version 1.3.13 for me without the extra logging?

**Comment by glk1001** [comment]
I've just created a create a clean version 1.3.13 without the extra logging.
